homeassistant/components/manual_mqtt/alarm_control_panel.py

```diff
@@ -347,7 +346,7 @@ def code_format(self) -> alarm.CodeFormat | None:
         """Return one or more digits/characters."""
         if self._code is None:
             return None
-        if isinstance(self._code, str) and re.search("^\\d+$", self._code):
+        if isinstance(self._code, str) and self._code.isdigit():
             return alarm.CodeFormat.NUMBER
         return alarm.CodeFormat.TEXT
```

Comment:
Seems `self._code` can also be a template. In that case `code_format` will always be `TEXT`. May be this is an existing bug.
I think this change is okay BTW.